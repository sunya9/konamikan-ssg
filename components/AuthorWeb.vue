<template>
  <div class="level is-mobile">
    <div class="level-left">
      <div v-if="author.website" class="level-item">
        <a :href="author.website" :class="color">
          <span class="icon" :class="size">
            <globe-icon />
          </span>
        </a>
      </div>
      <div v-if="author.facebook" class="level-item">
        <a :href="`https://facebook.com/${author.facebook}`" :class="color">
          <span class="icon" :class="size">
            <facebook-icon />
          </span>
        </a>
      </div>
      <div v-if="author.twitter" class="level-item">
        <a :href="`https://twitter.com/${author.twitter}`" :class="color">
          <span class="icon" :class="size">
            <twitter-icon />
          </span>
        </a>
      </div>
    </div>
  </div>
</template>
<script lang="ts">
import { Vue, Component, Prop } from 'nuxt-property-decorator'
import { Author } from '@tryghost/content-api'
import { GlobeIcon, TwitterIcon, FacebookIcon } from 'vue-feather-icons'

@Component({
  components: { GlobeIcon, TwitterIcon, FacebookIcon }
})
export default class AuthorWeb extends Vue {
  @Prop({ type: Object, required: true })
  author!: Author

  @Prop({ type: String, required: false, default: '' })
  color!: string

  @Prop({ type: String, required: false, default: 'is-small' })
  size!: string
}
</script>
