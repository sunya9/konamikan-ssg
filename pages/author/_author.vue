<template>
  <div>
    <app-header
      home-button
      :title="author.name"
      :description="author.bio"
      :cover="$resolveUrl(author.cover_image)"
    >
      <div class="container has-text-centered">
        <div class="media">
          <div class="media-left">
            <div class="image is-128x128">
              <img
                class="is-rounded"
                :src="author.profile_image"
                width="256"
                height="256"
                loading="lazy"
              />
            </div>
          </div>
          <div class="media-content">
            <div class="content">
              <h1 class="title is-1 has-text-white">
                {{ author.name }}
              </h1>
              <h2
                v-if="author.location"
                class="subtitle has-text-white-ter is-6"
              >
                {{ author.location }}
              </h2>
              <p class="has-text-white-ter">
                {{ author.bio }}
              </p>
              <author-web
                :author="author"
                color="has-text-white"
                size="is-normal"
              />
            </div>
          </div>
        </div>
      </div>
    </app-header>
    <main class="section">
      <div class="container">
        <posts-grouped-by-year-wrapper :posts="posts" />
      </div>
    </main>
  </div>
</template>
<script lang="ts">
import Vue from 'vue'
import { Component } from 'nuxt-property-decorator'
import { Author, AuthorsObject, PostObject } from '@tryghost/content-api'
import PostsGroupedByYearWrapper from '~/components/PostsGroupedByYearWrapper.vue'
import AppHeader from '~/components/AppHeader.vue'
import AuthorWeb from '~/components/AuthorWeb.vue'
import { reducePostFieldMapper, PostOrPageLight } from '~/util/util'

@Component({
  components: {
    PostsGroupedByYearWrapper,
    AppHeader,
    AuthorWeb
  },
  async asyncData({ $http, route, error }) {
    const [postsWrapper, authorsWrapper] = await Promise.all([
      $http.$get<PostObject>('/resources/posts'),
      $http.$get<AuthorsObject>('/resources/authors')
    ])
    const posts = postsWrapper.posts
    const authors = authorsWrapper.authors
    const author = authors.find((author) => author.slug === route.params.author)
    if (!author) return error({ statusCode: 404 })
    return {
      posts: posts
        .filter((post) =>
          post.authors?.some((postAuthor) => postAuthor.slug === author.slug)
        )
        .map(reducePostFieldMapper),
      author
    }
  }
})
export default class AuthorPage extends Vue {
  author!: Author
  posts!: PostOrPageLight[]
  head() {
    const meta: object[] = [
      {
        hid: 'description',
        property: 'description',
        content: this.author.bio
      },
      ...this.$createBaseMetadata(this.author)
    ]
    return {
      title: this.author.meta_title || this.author.name,
      meta
    }
  }
}
</script>
