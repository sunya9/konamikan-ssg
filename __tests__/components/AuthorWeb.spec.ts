import { mount } from '@vue/test-utils'
import { author } from '@/__tests__/fixtures/author'
import AuthorWeb from '@/components/AuthorWeb.vue'

describe('AuthorWeb', () => {
  it('normal', () => {
    expect(
      mount(AuthorWeb, {
        propsData: {
          author
        }
      })
    ).toMatchSnapshot()
  })

  it('override text color', () => {
    expect(
      mount(AuthorWeb, {
        propsData: {
          author,
          color: 'has-text-white'
        }
      })
    ).toMatchSnapshot()
  })

  it('override size', () => {
    expect(
      mount(AuthorWeb, {
        propsData: {
          author,
          size: 'is-normal'
        }
      })
    ).toMatchSnapshot()
  })
})
