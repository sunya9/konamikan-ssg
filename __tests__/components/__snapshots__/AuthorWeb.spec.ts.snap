// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorWeb normal 1`] = `
<div class="level is-mobile">
  <div class="level-left">
    <!---->
    <!---->
    <!---->
  </div>
</div>
`;

exports[`AuthorWeb override size 1`] = `
<div class="level is-mobile">
  <div class="level-left">
    <!---->
    <!---->
    <!---->
  </div>
</div>
`;

exports[`AuthorWeb override text color 1`] = `
<div class="level is-mobile">
  <div class="level-left">
    <!---->
    <!---->
    <!---->
  </div>
</div>
`;
